lib/elixir/lib/task/supervised.ex

```diff
@@ -379,15 +371,22 @@ defmodule Task.Supervised do
         ref = Process.monitor(pid)
         timer_ref = Process.send_after(self(), {:timeout, {monitor_ref, ref}}, timeout)
         send(parent_pid, {:spawned, {monitor_ref, position}, pid})
-        running_tasks = Map.put(running_tasks, ref, {position, type, pid, timer_ref})
+        task_info = %{
+          position: position,
+          type: type,
+          pid: pid,
+          timer_ref: timer_ref,
+          timed_out?: false,
```

Comment:
Instead of a boolean `timed_out?`, have it as `exit: :down` and then you set it to `:timed_out` in case of timeouts. What do you think?